Components/Queues/Drivers/PhpAmqpLibDriver.php

```diff
@@ -0,0 +1,342 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Smartbox\Integration\FrameworkBundle\Components\Queues\Drivers;
+
+use PhpAmqpLib\Channel\AMQPChannel;
+use PhpAmqpLib\Connection\AbstractConnection;
+use PhpAmqpLib\Connection\AMQPStreamConnection;
+use PhpAmqpLib\Exception\AMQPIOException;
+use PhpAmqpLib\Message\AMQPMessage;
+use PhpAmqpLib\Wire\AMQPTable;
+use Smartbox\Integration\FrameworkBundle\Components\Queues\QueueMessage;
+use Smartbox\Integration\FrameworkBundle\Components\Queues\QueueMessageInterface;
+use Smartbox\Integration\FrameworkBundle\Core\Messages\Context;
+use Smartbox\Integration\FrameworkBundle\DependencyInjection\Traits\UsesSerializer;
+use Smartbox\Integration\FrameworkBundle\Service;
+
+/**
+ * Class PhpAmqpLibDriver.
+ */
+class PhpAmqpLibDriver extends Service implements AsyncQueueDriverInterface
+{
+    use UsesSerializer;
+
+    /**
+     * This field specifies the prefetch window size in octets.
+     * The server will send a message in advance if it is equal to or smaller in size than the available prefetch size
+     * (and also falls into other prefetch limits).
+     * May be set to zero, meaning "no specific limit", although other prefetch limits may still apply.
+     * The prefetch-size is ignored if the no-ack option is set.
+     */
+    const PREFETCH_SIZE = null;
+
+    /**
+     * Represents the amount of message to consume by iteration
+     */
+    const PREFETCH_COUNT = 10;
```

Comment:
This value should be configured by driver, with this if we wanted to have some specific queue fetching more messages we could do it

Reply:
Yes. The constant is just a try to set a default param for this value. In any case, it's optional.

Reply:
Right now this value is not optional, it is the default value and always is 10, mu suggestion is to allow the user to define this when setting up the driver

Reply:
Can we add this as value to define in the Configuration of the bundle, so that if once in production we need to adjust this value we don't need to do a change in the bundle?

Reply:
There's a new branch to add parameters in queue drivers (params/amqp).

Reply:
Branch merged.